src/GameStatistics.ts

```diff
@@ -0,0 +1,276 @@
+import * as Unit from './entity/Unit';
+import * as Cards from './cards';
+import * as Achievements from './Achievements';
+import * as storage from "./storage";
+import Underworld from './Underworld';
+import { LAST_LEVEL_INDEX } from './config';
+
+//
+
+// Underworld Stats are only stored once, and we don't care "when" they happened
+export const underworldStats: IUnderworldStats = EmptyGlobalStatistics();
+export interface IUnderworldStats {
+  bestSpell: {
+    unitsKilled: number,
+    spell: string[]
+  };
+  longestSpell: string[];
+  // Used to make sure trackEndLevel() doesn't run logic multiple times
+  levelsComplete: number,
+  gameTimeElapsed: number,
+  runStartTime: number,
+  runWinTime: number | undefined,
+  runEndTime: number | undefined,
+}
+// This function can be used to initialize or reset globalStats
+export function EmptyGlobalStatistics(stats?: IUnderworldStats): IUnderworldStats {
+  return Object.assign(stats || {}, {
+    bestSpell: { unitsKilled: 0, spell: [] },
+    longestSpell: [],
+    levelsComplete: 0,
+    gameTimeElapsed: 0,
+    runStartTime: Date.now(),
+    runWinTime: undefined,
+    runEndTime: undefined,
+  })
+}
+
+//
+
+export enum StatDepth {
+  LIFETIME,
+  RUN,
+  LEVEL,
+  SPELL,
+}
+// Game Stats help us keep track of game events as they progress
+// and are separated by when they occured for the purpose of achievements
+export const gameStats: IStatistics[] = [
+  EmptyStatistics(), // 0 - Lifetime
+  EmptyStatistics(), // 1 - Run
+  EmptyStatistics(), // 2 - Level
+  EmptyStatistics(), // 3 - Spell
+];
+export interface IStatistics {
+  unitDeaths: number;
+  myPlayerDamageTaken: number;
+  myPlayerDeaths: number;
+  cardsCast: number;
+  myPlayerArrowsFired: number;
+  myPlayerCursesPurified: number;
```

Comment:
We're tracking `myPlayerCursesPurified` but we're not querying it anywhere, let's remove it and its corresponding function